src/codegen/compile.ml

```diff
@@ -9292,7 +9322,7 @@ let powInt64_shortcut fast env get_a get_b slow =
 
 (* kernel for Int64 arithmetic, invokes estimator for fast path *)
 let compile_Int64_kernel env name op shortcut =
-  Func.share_code2 env (prim_fun_name Type.Int64 name)
+  Func.share_code2 Func.Always env (prim_fun_name Type.Int64 name)
```

Comment:
I guess we could also inline these primitives, but they further increase the code size?

Reply:
Only if used. So yeah, we could inline these and some more probably

Reply:
Should I just merge as is and we can revisit later?

Reply:
Sure, this is already pretty good.